### PR TITLE
Meta: Standardize registration of CSS-only features

### PR DESCRIPTION
**Pull request by yakov116**
@fregante `clean-pinned-issues` was missing `awaitDomReady` so I we need to fix that before the next release.

### CODE REVIEW

**Comment by fregante** [comment]
Good, but further repetition can be avoided with a new `features.addCssFeature(FeatureID, BooleanFunction[])`

Inside it, it just calls features.add. 

This would be a shortcut for actual CSS-only features, and not for features that have multiple loaders.

**Comment by fregante** [comment]
No rush for the release by the way, I disabled the cronjob

File: source/features/index.tsx [hunk 1]
```diff
@@ -244,6 +248,19 @@ const add = async (id: FeatureID, ...loaders: FeatureLoader[]): Promise<void> =>
...
+const addCssFeature = async (id: FeatureID, loaders: CssInternalRunConfig): Promise<void> => {
+	async function cssOnlyFeature(featureName: string): Promise<void> {
+		await Promise.resolve(); // The event fires a bit before the document body loaded
+		document.body.classList.add('rgh-' + featureName);
+	}
+
+	void add(id, {
+		...loaders,
+		awaitDomReady: false,
+		init: () => void cssOnlyFeature(id)
```
Comment (fregante) [inline source/features/index.tsx:260]: Just inline it, don’t call a secondary function.

Comment (yakov116) [reply source/features/index.tsx:260]: Forgot to make it a draft. I did not have time to finish it yet.

**Comment by yakov116** [comment]
@fregante if you want to release sooner I have no issue if you want finish this up. If not I will do it a drop later.

**Comment by yakov116** [comment]
Need typescript help

**Comment by fregante** [comment]
The whole implementation is wrong, it’s not just the types. The function accepts an array but you’re using the rest operator onto an object. And also you’re passing single functions instead of arrays of functions

**Comment by yakov116** [comment]
> And also you’re passing single functions instead of arrays of functions

I am not sure what you mean here

**Comment by fregante** [comment]
The parameter type was `BooleanFunction[]`
you passed `BooleanFunction`

You fixed it

File: source/features/align-issue-labels.tsx [hunk 2]
```diff
@@ -3,14 +3,4 @@ import * as pageDetect from 'github-url-detection';
...
-}
-
-void features.add(__filebasename, {
-	awaitDomReady: false,
-	include: [
-		pageDetect.isConversationList
-	],
-	init
-});
+void features.addCssFeature(__filebasename,	[pageDetect.isConversationList]);
```
Comment (fregante) [inline source/features/align-issue-labels.tsx:6]: There’s a tab instead of a space on this line in every file

File: source/features/index.tsx [hunk 3]
```diff
@@ -244,6 +244,18 @@ const add = async (id: FeatureID, ...loaders: FeatureLoader[]): Promise<void> =>
 	}
 };
 
+const addCssFeature = async (id: FeatureID, pageDetections: BooleanFunction[]): Promise<void> => {
+	void add(id, {
+		include: [
+			...pageDetections
```
Comment (fregante) [inline source/features/index.tsx:250]: This variable should be called `include` and since it’s already an array you don’t need to use the rest parameter

**Review by fregante — CHANGES_REQUESTED** [review]

File: source/refined-github.ts [hunk 4]
```diff
@@ -24,18 +24,25 @@ import './features/align-repository-header.css';
 // DO NOT add CSS files here if they are part of a JavaScript feature.
 // Import the `.css` file from the `.tsx` instead.
 
+// CSS ONLY features
```
Comment (fregante) [inline source/refined-github.ts:27]:
```suggestion
// CSS-only disableable features
```

**Comment by fregante** [comment]
Excellent!

**Pull request merged**